src/vehicle_use.cpp

```diff
@@ -1130,6 +1130,33 @@ void vehicle::open_all_at( int p )
     }
 }
 
+bool vehicle::turn_on_internal_lights( bool dome_only )
+{
+    bool success = false;
+
+    const auto turn_on_light_type = [&]( vpart_bitflags light_type ) {
+        bool switched = false;
+        auto light_parts = get_parts( light_type );
+        for( vehicle_part *e  : light_parts ) {
+            e->enabled = true;
+            switched = true;
+        }
+        return switched;
+    };
+
+    if( dome_only ) {
+        success = turn_on_light_type( VPFLAG_DOME_LIGHT );
+    } else {
+        success |= turn_on_light_type( VPFLAG_AISLE_LIGHT ) | turn_on_light_type( VPFLAG_ATOMIC_LIGHT );
```

Comment:
Just to be sure, you intentionally are not turning on the dome light as well here?

Reply:
Right, since that'd make two things for the player to turn off later, despite the dome lights not being helpful.

Reply:
Oh nevermind, so the `get_parts` is checking if the player is actually sitting on the controls while crafting.  This still doesn't address some of the problems above though.

Reply:
`vehicle::turn_on_internal_lights()` returns `true` if a light was actually turned on.  You're right that it is possible that the player is also not inside the light area of an aisle light after turning the lights on though.  In such a case, the activity will cancel on the next turn, because `tried_turning_on_lights` is set to true this time.

As for checking if the player can reach the controls, I was abstracting away the whole "walk to the controls" part.  Turning on the lights does cost 300 moves though, so it isn't free.  Maybe it should be higher?

Reply:
Player is still able to turn on the lights without actually being able to reach them.  Handing waving this away is not ok IMO.  Just subtracting a bunch of moves is not ok IMO.

You're still not actually checking if the player is *in* the light now.  Just because you turned on a bunch of lights doesn't mean any of them helped the player be able to craft.